packages/website/docs/templates.md

```diff
@@ -0,0 +1,221 @@
+---
+id: templates
+title: Displaying items with Templates
+---
+
+Templates let you customize the display of your autocomplete's items.
+
+Once you've set up your [data sources](sources), you need to define how they display in your autocomplete experience. It encompasses the structure for each item and the way they look.
+
+Autocomplete provides a Templates API to let you fully customize the render of each item.
+
+## Usage
+
+### Rendering each item
+
+The rendering system of Autocomplete uses an agnostic virtual DOM implementation. You can return anything from each template as long as they're valid virtual DOM elements (VNodes).
+
+For example, templates can return a string:
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources() {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return `Result: ${item.name}`;
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+Or a [Preact](https://preactjs.com/) component:
+
+```js
+/** @jsx h */
+import { h } from 'preact';
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources() {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return <div>{item.name}</div>;
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
```

Comment:
Perhaps add a mention that the default Autocomplete rendering is based on Preact 10, so previous versions are not compatible.

Reply:
Done in d638ebd.